src/SwedbankPay.Sdk.Infrastructure/JsonSerialization/TypedSafeEnumValueConverter.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace SwedbankPay.Sdk.JsonSerialization
+{
+    public class TypedSafeEnumValueConverter<TEnum, TValue> : JsonConverter<TEnum>
+        where TEnum : TypeSafeEnum<TEnum, TValue>
+    {
+        /// <summary>
+        /// </summary>
+        /// <param name="reader"></param>
+        /// <param name="typeToConvert"></param>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        public override TEnum Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if(reader.TokenType == JsonTokenType.PropertyName)
+            {
+                reader.GetString();
+                reader.Read();
+            }
+
+            if(reader.TokenType == JsonTokenType.StartObject)
+            {
+                reader.Read();
+                var valueToReturn = Read(ref reader, typeToConvert, options);
+                reader.Read();
+                return valueToReturn;
+            }
+
+            try
+            {
+                TValue value;
+                if (reader.TokenType == JsonTokenType.Number && typeof(TValue) != typeof(long) && typeof(TValue) != typeof(bool))
+                    value = (TValue)Convert.ChangeType(reader.GetInt64(), typeof(TValue));
+                else

```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'else' block.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-dotnet/pullRequest?prid=6290774)